src/globus_sdk/types.py

```diff
@@ -0,0 +1,12 @@
+from array import array
+from typing import Union
+
+from typing_extensions import Protocol
+
+# From: https://docs.python.org/3/glossary.html#term-bytes-like-object
+BytesLike = Union[array, bytes, bytearray, memoryview]
```

Comment:
I don't see this being used anywhere. Am I missing it?

`bytes` are often returned or required by python APIs -- DBs, wsgi, etc. -- so there's some definite value in supporting that type for string-like parameters. But these other types are more obscure. I'm not sure we want to worry about supporting them.